valhalla/mjolnir/osmnodelinguistic.h

```diff
@@ -0,0 +1,345 @@
+    return ref_pronunciation_katakana_lang_index_;
+  }
+
+  /**
+   * Sets the index for the ref jeita pronunciation
+   * @param  idx  Index for the reference jeita pronunciation.
+   */
+  void set_ref_pronunciation_jeita_index(const uint32_t idx) {
+    ref_pronunciation_jeita_index_ = idx;
+  }
+
+  /**
+   * Get the ref jeita pronunciation index.
+   * @return  Returns the index for the ref jeita pronunciation.
+   */
+  uint32_t ref_pronunciation_jeita_index() const {
+    return ref_pronunciation_jeita_index_;
+  }
+
+  /**
+   * Sets the index for the ref jeita lang pronunciation
+   * @param  idx  Index for the reference jeita lang pronunciation.
+   */
+  void set_ref_pronunciation_jeita_lang_index(const uint32_t idx) {
+    ref_pronunciation_jeita_lang_index_ = idx;
+  }
+
+  /**
+   * Get the ref jeita pronunciation lang index.
+   * @return  Returns the index for the ref jeita lang pronunciation.
+   */
+  uint32_t ref_pronunciation_jeita_lang_index() const {
+    return ref_pronunciation_jeita_lang_index_;
+  }
+
+  /**
+   * Has any index been set?
+   * @return  True is nothing is set
+   */
+  bool isEmpty() const {
+    return (name_lang_index_ == 0 && name_pronunciation_ipa_index_ == 0 &&
+            name_pronunciation_ipa_lang_index_ == 0 && name_pronunciation_nt_sampa_index_ == 0 &&
+            name_pronunciation_nt_sampa_lang_index_ == 0 && name_pronunciation_katakana_index_ == 0 &&
+            name_pronunciation_katakana_lang_index_ == 0 && name_pronunciation_jeita_index_ == 0 &&
+            name_pronunciation_jeita_lang_index_ == 0 && ref_lang_index_ == 0 &&
+            ref_pronunciation_ipa_index_ == 0 && ref_pronunciation_ipa_lang_index_ == 0 &&
+            ref_pronunciation_nt_sampa_index_ == 0 && ref_pronunciation_nt_sampa_lang_index_ == 0 &&
+            ref_pronunciation_katakana_index_ == 0 && ref_pronunciation_katakana_lang_index_ == 0 &&
+            ref_pronunciation_jeita_index_ == 0 && ref_pronunciation_jeita_lang_index_ == 0);
+  }
+
+  // pronunciations / langs
+  uint32_t name_lang_index_;
+  uint32_t name_pronunciation_ipa_index_;
+  uint32_t name_pronunciation_ipa_lang_index_;
+  uint32_t name_pronunciation_nt_sampa_index_;
+  uint32_t name_pronunciation_nt_sampa_lang_index_;
+  uint32_t name_pronunciation_katakana_index_;
+  uint32_t name_pronunciation_katakana_lang_index_;
+  uint32_t name_pronunciation_jeita_index_;
+  uint32_t name_pronunciation_jeita_lang_index_;
+  uint32_t ref_lang_index_;
+  uint32_t ref_pronunciation_ipa_index_;
+  uint32_t ref_pronunciation_ipa_lang_index_;
+  uint32_t ref_pronunciation_nt_sampa_index_;
+  uint32_t ref_pronunciation_nt_sampa_lang_index_;
+  uint32_t ref_pronunciation_katakana_index_;
+  uint32_t ref_pronunciation_katakana_lang_index_;
+  uint32_t ref_pronunciation_jeita_index_;
+  uint32_t ref_pronunciation_jeita_lang_index_;
+};
+} // namespace mjolnir
+} // namespace valhalla
+
+#endif // VALHALLA_MJOLNIR_OSMLINGUISTIC_H
```

Comment:
```suggestion
#pragma once

#include <cstdint>

namespace valhalla {
namespace mjolnir {

/**
 * OSM node linguistic information.
 */
struct OSMNodeLinguistic {
  static bool is_empty(const OSMNodeLinguistic& maybe_empty) {
    static OSMNodeLinguistic definitely_empty{};
    return maybe_empty == definitely_empty;
  }

  uint32_t name_lang_index_;
  uint32_t name_pronunciation_ipa_index_;
  uint32_t name_pronunciation_ipa_lang_index_;
  uint32_t name_pronunciation_nt_sampa_index_;
  uint32_t name_pronunciation_nt_sampa_lang_index_;
  uint32_t name_pronunciation_katakana_index_;
  uint32_t name_pronunciation_katakana_lang_index_;
  uint32_t name_pronunciation_jeita_index_;
  uint32_t name_pronunciation_jeita_lang_index_;
  uint32_t ref_lang_index_;
  uint32_t ref_pronunciation_ipa_index_;
  uint32_t ref_pronunciation_ipa_lang_index_;
  uint32_t ref_pronunciation_nt_sampa_index_;
  uint32_t ref_pronunciation_nt_sampa_lang_index_;
  uint32_t ref_pronunciation_katakana_index_;
  uint32_t ref_pronunciation_katakana_lang_index_;
  uint32_t ref_pronunciation_jeita_index_;
  uint32_t ref_pronunciation_jeita_lang_index_;
};

} // namespace mjolnir
} // namespace valhalla
```

Reply:
i would humbly ask you to consider a much more succinct implementation for this basic struct to keep it a POD and just assign to and read from the fields directly. in this case zero'ing the struct is a simple as using the brace initializer `{}`. if we upgraded to c++20 we could even have a much more succinct equality check but i guess we'll save that for later

Reply:
also if you'd rather not consider it, no worries

Reply:
Save for later.  ;-)  Unfortunately, I don't have time to retest everything right now.